datajoint/schema.py

```diff
@@ -87,15 +87,18 @@ def process_relation_class(relation_class, context):
 
         process_relation_class(cls, context=self.context)
 
-        # Process subordinate relations
-        parts = list()
-        is_part = lambda x: inspect.isclass(x) and issubclass(x, Part)
+        # Process part relations
+        def is_part(x):
+            return inspect.isclass(x) and issubclass(x, Part)
 
-        for var, part in inspect.getmembers(cls, is_part):
-            parts.append(part)
-            part._master = cls
-            # TODO: look into local namespace for the subclasses
-            process_relation_class(part, context=dict(self.context, **{cls.__name__: cls}))
+        parts = list()
+        for part in dir(cls):
```

Comment:
Please correct me if I am wrong, but I think `inspect.getmembers` actually evaluates every member.  So if there is a computed property, e.g. `_populated_from`, it will actually trigger its computation.  In contrast, the proposed implementation looks at the names first and only evaluates members if they start with a capital letter.  So it avoids extra computations.